src/bootstrap/builder.rs

```diff
@@ -774,7 +774,17 @@ impl<'a> Builder<'a> {
 
         cargo
             .env("CARGO_TARGET_DIR", out_dir)
-            .arg(cmd);
+            .arg(cmd)
+            .arg("-Zconfig-profile");
+
+        let profile_var = |name: &str| {
+            let profile = if self.config.rust_optimize {
```

Comment:
unrelated note: we should probably drop rust_optimize as a config flag entirely since there's no real point in a non-optimized build I imagine